HIP/hip-991.md

```diff
@@ -44,182 +44,310 @@ This document outlines the development of a fixed fee system for the submission
+**Related Requests:**
 
-## **Specification**
+This enhancement is in response to community feedback (Tier Bot) requesting more versatile and predictable economic models within topic IDs on the Hedera network.
 
-### **Conceptual Overview**
+## User personas and stories
 
-We propose adding a fixed fee mechanism to the Hedera Consensus Service (HCS) for topic messages, similar to the custom fee structures in the Hedera Token Service (HTS). This will involve creating new Protobuf messages and modifying existing ones to accommodate fixed fees for topics.
+**User Personas:**
 
-### **Protobuf Messages Needed**
+* **Topic Operator:** Entities or individuals who manage and operate topics, interested in stable revenue and easy management of topic-related transactions.
+* **Topic Operator:** Entities or individuals who manage and operate topics, interested in reducing the noise of public topic IDs/making it more difficult to spam an open topic.
+* **Regular User:** End users who interact with topics, adding paid methods of interaction and submission to topics.
 
-### **FixedFee**
+**User Stories:**
 
-The `FixedFee` message represents a fixed fee charged for each message submission to a topic. This protobuf message already exists so it can be re-used.
+* As a **Topic Operator**, I want to set a fixed fee for submitting messages to my topic, so that I can ensure a consistent revenue stream and manage my topic effectively.
+* As a **Regular User**, I want to know the exact cost of interacting with a topic.
 
-```
-message FixedFee {
-  // The number of units to assess as a fee
-  int64 amount = 1;
-  // The denomination of the fee; taken as hbar if left unset
-  TokenID denominating_token_id = 2;
-}
-```
+## Specification
 
-### **CustomFixedFee**
+### Overview
 
-The `CustomFixedFee` message defines the type of fee and the account receiving the fee assessed during a message submission to the associated topic. A custom fee may only be fixed and must specify a fee collector account to receive the assessed fees.
+We propose adding a fixed fee mechanism to the Hedera Consensus Service (HCS) for topic messages, similar to the custom fee structures in the Hedera Token Service (HTS). This will involve creating new protocol buffers (protobuf) messages and modifying existing ones to accommodate fixed fees for topics.
 
-```
-message CustomFixedFee {
-  // Fixed fee to be charged
-  FixedFee fixed_fee = 1;
+### Requirements and Scope
 
-  // The account to receive the custom fee
-  AccountID fee_collector_account_id = 3;
-}
-```
+* Topics can have a Fee Schedule Key set at creation time.
+* Fee Schedule Key can manage fee updates and send free messages to the topic.
+* The Fee Schedule Key can be updated with an update topic transaction signed by the Admin Key and the new Fee Schedule Key.
```

Comment:
Should we add the ability for the fee schedule key to update itself? I think a HIP went through recently that gave lower level privilege keys that ability

Reply:
Yep, true, that makes sense! It was the [HIP-540](https://hips.hedera.com/hip/hip-540) that added the following rules to HTS:

> 1. Only the admin key should be able to remove itself or other keys.
> 2. All keys can change themselves to another valid or unusable key (such as all-zeros). Except for the admin key - we do not support setting an unusable admin key. This is because removing the admin key and making it unusable are functionally identical. It should simply be removed if no longer desired.

I'll send an update about this point and add a reference to the HIP-540.

Reply:
The text now reflects the change, let me know if it makes sense.